Form1.cs

```diff
@@ -26,51 +27,80 @@ private void Form1_Load(object sender, EventArgs e)
-            Validation2 valida = new Validation2
+            try
             {
-               Nome = txtNome.Text.Trim(),
-               Endereco = txtEndereco.Text.Trim(),
-               Cep = txtCep.Text.Trim(),
-               Cidade = txtCidade.Text.Trim(),
-               Estado = cboEstado.Text.Trim(),
-               IdEstado = Convert.ToInt32(cboEstado.SelectedValue),
-               EstadoCivil = cboEstadoCivil.Text.Trim(),
-               IdEstadoCivil = Convert.ToInt32(cboEstadoCivil.SelectedValue)
-            };
+               cliente.Validar();
 
-            valida.IsertClient();
+               Validation2 valida = new Validation2
+               {
+                  Nome = txtNome.Text.Trim(),
+                  Endereco = txtEndereco.Text.Trim(),
+                  Cep = txtCep.Text.Trim(),
+                  Cidade = txtCidade.Text.Trim(),
+                  Estado = cboEstado.Text.Trim(),
+                  IdEstado = Convert.ToInt32(cboEstado.SelectedValue),
+                  EstadoCivil = cboEstadoCivil.Text.Trim(),
+                  IdEstadoCivil = Convert.ToInt32(cboEstadoCivil.SelectedValue)
+               };
 
-            dgClientes.PreencheDataGrid();
-         }
-         catch (Exception x)
-         {
-            status = x.Message;
-            if (cliente.txtControle == "nome") txtNome.Focus();
-            if (cliente.txtControle == "endereco") txtEndereco.Focus();
-            if (cliente.txtControle == "cidade") txtCidade.Focus();
-            if (cliente.txtControle == "cep") txtCep.Focus();
-            if (cliente.txtControle == "estado") cboEstado.Focus();
-            if (cliente.txtControle == "estadoCivil") cboEstadoCivil.Focus();
+               valida.IsertClient();
+
+               dgClientes.PreencheDataGrid();
+            }
+            catch (Exception x)
+            {
+               status = x.Message;
+               if (cliente.txtControle == "nome") txtNome.Focus();
+               if (cliente.txtControle == "endereco") txtEndereco.Focus();
+               if (cliente.txtControle == "cidade") txtCidade.Focus();
+               if (cliente.txtControle == "cep") txtCep.Focus();
+               if (cliente.txtControle == "estado") cboEstado.Focus();
+               if (cliente.txtControle == "estadocivil") cboEstadoCivil.Focus();
+               cliente.txtControle = "";
+            }
+            finally
+            {
+               lblStatus.Text = "Status: " + status;
+            }
          }
-         finally
+         else
          {
-            lblStatus.Text = "Status: " + status;
+            cmdIncluir.Text = "&Salvar";
+            txtNome.Focus();
          }
       }
+
+      private void cmdFechar_Click(object sender, EventArgs e)
+      {
+         this.Close();
+      }
+
+      private void dgClientes_CellDoubleClick(object sender, DataGridViewCellEventArgs e)
+      {
+         PreencheFormCliente(Convert.ToInt32(dgClientes.Rows[e.RowIndex].Cells[0].Value));
```

Comment:
Dar nomes significativos às variaveis

```suggestion
         int idClienteSelecionado = Convert.ToInt32(dgClientes.Rows[e.RowIndex].Cells[0].Value);
         PreencheFormCliente(idClienteSelecionado);
```